parrot/src/pfs_table.cc

```diff
@@ -1137,7 +1171,7 @@ int pfs_table::access( const char *n, mode_t mode )
 	pfs_name pname;
 	int result = -1;
 
-	if(resolve_name(0,n,&pname)) {
+	if(resolve_name(0,n,&pname,X_OK)) {
 		result = pname.service->access(&pname,mode);
```

Comment:
Also, the mode passed to resolve_name will need to be OR'd with the mode passed to `pname.service->access`.